### DISCUSSION TITLE
How to contribute?

### DISCUSSION THREAD

**Discussion by pradkrish** [stdgraph/graph-v2#31] [General]
Hello, my name is Pradeep and I work as a Computer Vision engineer in Sweden. I am a professional C++ developer with 5 years of experience writing modern C++. I enjoyed the presentation on cppcon 2022 and I would really like to contribute code to the project. As I am new to cpp20, I am happy to work on small implementations and build my way up. any suggestions on how/where to start? Thanks.

**Comment by pratzl** [comment]
Thanks for reaching out. I think the best way is to pull the code and build it initially. Initial impressions are always very important, so anything you have to offer would be valuable, keeping in mind that it's still early and a little rough. I'm discussing with others on the team how we can include others like you, so I ask for your patience. Can you help me understand your experience with C++20 and graph algorithms, so I understand how you might fit in what we're doing?

**Reply by pradkrish** [reply]
Thank you for getting back. Yeah, I am currently browsing through the code to understand how it's structured and use example(s) to understand how to use the library. I went through the documentation in nwgraph and it was very useful. 

I have done a few tutorials on the main features of C++20 but I have no experience working with a C++20 project. I am learning by browsing through the code. My knowledge of graph theory comes from university courses on Data structures and algorithms. I have also played around for a while with igraph library and I have also made a few code contributions there. I can see there is a lot to be done with this library. I find it exciting and challenging and I hope to be able to help out.

**Comment by pratzl** [comment]
There's certainly a lot that remains to be done. The big area we need work in is the algorithms. Adding examples using the existing functionality would be useful, documentation is needed, and other ideas are welcome. The C++20 features we're using most are ranges and concepts, so it might be a good idea to find something to get more experience with that. Is there something that you'd like to do?

**Reply by pradkrish** [reply]
Thanks, I am currently looking into implementing triangle count, which will hopefully help me to make contributions to algorithms. I will use https://github.com/pnnl/NWGraph/blob/master/include/nwgraph/algorithms/triangle_count.hpp as reference.

**Reply by pratzl** [reply]
That sounds good to me. Once you have something ready you can do the pull request. We're just getting started on this, so you may be the person we learn with to figure out how to communicate, and what we want.
